alacritty/src/renderer/mod.rs

```diff
@@ -807,7 +808,11 @@ impl<'a> RenderApi<'a> {
     }
 
     #[inline]
-    fn add_render_item(&mut self, cell: &RenderableCell, glyph: &Glyph) {
+    fn add_render_item(&mut self, cell: &mut RenderableCell, glyph: &Glyph) {
+        cell.bg_alpha *= 255.;
+        if !self.config.colors.opaque_background_colors {
+            cell.bg_alpha *= self.config.ui_config.background_opacity();
+        }
```

Comment:
Is this the right place for this? If `cell` already has a `bg_alpha` field, shouldn't this property be applied when this field is initialized? Doing it right here seems somewhat arbitrary?

Reply:
Well, I think I basically followed https://github.com/alacritty/alacritty/pull/4196#issuecomment-686057677 with where to put the scaling. Maybe there is a better way to put it, since I am anyway going to retrace my steps to address the other options I'll bear this is in mind an see if there is a more logical place.

Reply:
Ah, I've previously said the following:



> > It does sound like the opacity handling should all happen in alacritty, but it would require refactoring to move the relevant bits from alacritty_terminal to alacritty first?
>
> Yes. Currently it is not done entirely in alacritty, which it should.

This is technically correct, but fact is that opacity handling happens in `alacritty_terminal` already. So if we want to move it outside of `alacritty_terminal`, it should probably all be moved. I believe that's probably what I was thinking of back then too. Generally though that's a lot of work, so I'd be fine with putting it in `alacritty_terminal` for now, you should be able to find the other alpha stuff in `RenderableCell::new` I think.

Reply:
I think I see where to do that in alacritty_terminal. However while I can get to `config.colors.opaque_background_color` I am not sure that `config.ui_config.background_opacity()` is accessible from there - at least the compiler keeps shouting at me? I suppose I could move the `background_opacity` to `color`, but given that you were keen on moving it to `window` I am not sure that is desirable (plus I am not sure how to support the previous location of that field)?

Reply:
Fair enough, I'd hold on this for now then I suppose.